lib/utmp.c

```diff
@@ -275,7 +280,8 @@ prepare_utmp(const char *name, const char *line, const char *host,
 	utent->ut_type = USER_PROCESS;
 	utent->ut_pid = main_pid;
 	STRNCPY(utent->ut_line, line);
-	if (NULL != ut) {
+	if (   (NULL != ut)
+	    && ('\0' != utent->ut_id[0])) {
```

Comment:
Why this added check?  Is is related to the other code changes, or is it an unrelated fix?

Reply:
Updated with correct variable used in the check.

Reply:
This is why I think comments are bad.